File: spark-doris-wb-quality-detect-etl/src/main/java/com/qtech/comparison/dpp/batch/BatchEngine.java
package com.qtech.comparison.dpp.batch;

/**
 * author :  gaozhilin
 * email  :  gaoolin@gmail.com
 * date   :  2022/06/12 10:07:53
 * desc   :
 */


public abstract class BatchEngine {

    protected abstract void srvProcessData() throws Exception;

    public void start() throws Exception {
        this.srvProcessData();
    }
}

File: spark-wb-olp-check-etl/src/main/scala/com/qtech/spotChecking/dpp/batch/BatchEngine.scala
package com.qtech.spotChecking.dpp.batch

/**
 * author :  gaozhilin
 * email  :  gaoolin@gmail.com
 * date   :  2024/04/11 15:31:18
 * desc   :
 */


abstract class BatchEngine {

  def start(): Unit =
    this.srvProcessData()

  def srvProcessData(): Unit
}
